accepted/2022/dotnet-nuget-why-proposal.md

```diff
@@ -118,6 +118,11 @@ dotnet nuget why [<PROJECT>|<SOLUTION>] <PACKAGE_NAME>
 
 dotnet nuget why -h|--help
 ```
+
+We will show the resolved versions of packages in the graph, rather than the versions requested by the top-level package.
+
+We will also deduplicate graphs in certain cases. If we have already shown a package's dependencies in the graph in another place, we will truncate the graph at that node and show a `(*)` symbol to indicate that the package's dependencies have already been shown. Further, if multiple target frameworks in a project have the same dependency graph, we will show a combined graph for them. (See [examples](#examples))
```

Comment:
I'm not really a fan of this `(*)` proposal. I looked at the screenshots before reading the text, and I couldn't guess what it meant. So, if we add it, I think it's very important that the program output tell customers what it means.

However, even if the subtree was already output in another part of the graph, I don't think that trimming it provides a big value. It will make the output shorter, sure, but it actually doesn't in the examples in your screenshots, becuase the subtree it trimmed was just one package. And even if it did make the output shorter, there's also the risk of making it harder to understand, since the customer will now need to look for other instances of the parent package, to see what the children packages actually are.

Reply:
This seems fair. Having to look through the rest of the graph to see the subtree isn't ideal, especially as default behavior. We can consider adding this as a CLI option in the future, but we'll show the complete graph for now.

I've moved the duplicate subtree trimming into the 'Future work' section now, as a potential CLI option (a boolean option allowing users to de-duplicate subtrees, or a verbosity option, etc.)